### PR TITLE
Use useradd on Alpine when user ID is large

### PR DESCRIPTION
**Pull request by DenisRumyantsev**
***Description:*** The `adduser` command can be used in Alpine by default without installing any additional packages. Therefore, this command has a higher priority. However, it does not work if the user ID is greater than `256000`. In this case, we should install the additional `shadow` package and use the `useradd` command instead.

The `shadow` package will be used in the following cases:
* if this package is preinstalled in the container;
* if user ID is outside the range of the `adduser` command.

[Related issue](https://github.com/microsoft/azure-pipelines-agent/issues/4519)

### CODE REVIEW

File: src/Agent.Worker/ContainerOperationProvider.cs [hunk 1]
```diff
@@ -708,7 +708,14 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
                     Func<string, string, string, string> addUserWithIdAndGroup;
                     Func<string, string, string> addUserToGroup;
 
-                    if (isAlpineBasedImage)
+                    bool userIdIsLarge = Int64.Parse(container.CurrentUserId) > 256000;
```
Comment (DergachevE) [inline src/Agent.Worker/ContainerOperationProvider.cs:711]: Can be some constant defined with meaningful name?

**Comment by KonstantinTyukalov** [comment]
@DenisRumyantsev can we just always use the `shadow` package? Does it have some side effects?

File: src/Agent.Worker/ContainerOperationProvider.cs (hunk 1 above)
Comment (KonstantinTyukalov) [inline src/Agent.Worker/ContainerOperationProvider.cs:711]: Do we always expect `CurrentUserId` will be a valid `int`?

Comment (DenisRumyantsev) [reply src/Agent.Worker/ContainerOperationProvider.cs]: Yes, user ID should be a valid `int`

File: src/Agent.Worker/ContainerOperationProvider.cs [hunk 2]
```diff
@@ -708,7 +708,14 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
...
+
+                    if (isAlpineBasedImage && userIdIsLarge)
+                    {
+                        await DockerExec(executionContext, container.ContainerId, "apk add shadow");
```
[the remark this answers is not in the document]
Comment (DenisRumyantsev) [reply src/Agent.Worker/ContainerOperationProvider.cs]: Fixed. Now we will use this command only if it is necessary.

File: src/Agent.Worker/ContainerOperationProvider.cs (hunk 1 above)
Comment (DenisRumyantsev) [reply src/Agent.Worker/ContainerOperationProvider.cs]: Fixed!

**Comment by DenisRumyantsev** [comment]
@KonstantinTyukalov I suggest we can use the `shadow` package always when it is already installed in the container, and don't use it if it is not necessary and it is not installed yet.

**Review by KonstantinTyukalov — APPROVED** [review]

**Review by DergachevE — APPROVED** [review]

**Pull request merged**